### PR TITLE
more support for DB constraints

### PR DESCRIPTION
**Pull request by jgaleotti**

### CODE REVIEW

File: core/src/test/kotlin/org/evomaster/core/database/extract/postgres/Ind0ExtractTest.kt [hunk 1]
```diff
@@ -44,4 +45,14 @@ class Ind0ExtractTest : ExtractTestBasePostgres() {
...
     }
+
+    @Test
+    fun testSqlInsert() {
+        val schema = SchemaExtractor.extract(connection)
+
+        val builder = SqlInsertBuilder(schema)
+//        val actions = builder.createSqlInsertionAction("x", setOf("p_at", "created_at"))
+//        val genes = actions[0].seeGenes()
+
```
Comment (arcuri82) [inline core/src/test/kotlin/org/evomaster/core/database/extract/postgres/Ind0ExtractTest.kt]: a new test, but commented out and no assertions?

File: dbconstraint/src/main/java/org/evomaster/dbconstraint/parser/jsql/JSqlConditionParser.java [hunk 2]
```diff
@@ -3,21 +3,49 @@
...
 import org.evomaster.dbconstraint.parser.SqlConditionParser;
 import org.evomaster.dbconstraint.parser.SqlConditionParserException;
 
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 public class JSqlConditionParser extends SqlConditionParser {
 
+    public static final String FORMULA_EQUALS_FORMULA_PATTERN = "\\(\\s*\\(([^<]*)\\)\\s*=\\s*\\(([^<]*)\\)\\s*\\)";
+
```
Comment (arcuri82) [inline dbconstraint/src/main/java/org/evomaster/dbconstraint/parser/jsql/JSqlConditionParser.java]: can you provide bit of explanation for this Regex? eg, what is for

File: dbconstraint/src/main/java/org/evomaster/dbconstraint/parser/jsql/JSqlConditionParser.java [hunk 3]
```diff
@@ -3,21 +3,49 @@
...
+    /**
+     * replaces unsupported grammar of JSQLParser with equivalent supported constructs
+     *
+     * @param originalSqlStr
+     * @return
+     */
+    private String transformDialect(String originalSqlStr) {
+        // replace " = ANY (ARRAY[...])" with " IN (...)"
+        return originalSqlStr.replaceAll("=\\s*ANY\\s*\\(\\s*ARRAY\\s*\\[([^<]*)\\]\\s*\\)", " IN ($1)");
+    }
```
Comment (arcuri82) [inline dbconstraint/src/main/java/org/evomaster/dbconstraint/parser/jsql/JSqlConditionParser.java]: can u provide a bit of explanation for it?

File: core/src/test/kotlin/org/evomaster/core/database/extract/postgres/Ind0ExtractTest.kt (hunk 1 above)
Comment (jgaleotti) [reply core/src/test/kotlin/org/evomaster/core/database/extract/postgres/Ind0ExtractTest.kt]: This is the test that creates the actions for the industrial case study. The schema is already extracted, but there are many column types (xml, jsonb, text) that are not currently supported by EvoMaster. How should I proceed on cases like this one? (i.e. I have the test case but is failing) Should I add an @ignore ?

File: dbconstraint/src/main/java/org/evomaster/dbconstraint/parser/jsql/JSqlConditionParser.java (hunk 2 above)
Comment (jgaleotti) [reply dbconstraint/src/main/java/org/evomaster/dbconstraint/parser/jsql/JSqlConditionParser.java]: Sure.

File: core/src/test/kotlin/org/evomaster/core/database/extract/postgres/Ind0ExtractTest.kt (hunk 1 above)
Comment (arcuri82) [reply core/src/test/kotlin/org/evomaster/core/database/extract/postgres/Ind0ExtractTest.kt]: just add a brief "TODO" comment :)

File: dbconstraint/src/main/java/org/evomaster/dbconstraint/parser/jsql/JSqlConditionParser.java (hunk 3 above)
Comment (jgaleotti) [reply dbconstraint/src/main/java/org/evomaster/dbconstraint/parser/jsql/JSqlConditionParser.java]: I enhanced the description and the procedure for transforming some JSQL unsupported constructs

**Comment by arcuri82** [comment]
made mistake of approving without checking CI. there are test failures for `JSqlConditionParserTest`

**Comment by jgaleotti** [comment]
I fixed the error in the JSqlConditionParser

**Comment by jgaleotti** [comment]
Sorry. My bad. I pushed the fixes.

Cheers
JP

**Pull request merged**